src/main/java/net/snowflake/ingest/streaming/SnowflakeStreamingIngestClientFactory.java

```diff
@@ -44,6 +47,17 @@ public Builder setParameterOverrides(Map<String, Object> parameterOverrides) {
       return this;
     }
 
+    /**
+     * Optional builder to enable JMX metrics for streaming client.
+     *
+     * @param enableJmxMetrics true if metrics per client needs to be emitted
+     * @return this builder instance
+     */
+    public Builder enableJmxMetrics(final boolean enableJmxMetrics) {
+      this.enableJmxMetrics = enableJmxMetrics;
+      return this;
+    }
+
```

Comment:
could you add everything side `parameterOverrides`?

Reply:
I am a bit hesitant about that because that is not a good interface for interaction from client perspective. 

`enableJmxMetrics` is descriptive enough to use. 

Infact I would rather not keep `setParameterOverrides` since it is inviting customers to reset defaults. I thought we were opposed to that. (Althought it is open source)

Reply:
This [doc](https://docs.google.com/document/d/1m-vMJjFwd0_uAWidk1tv0DDp-HUdZqBCe5rZjHdFUzA/edit) that @sfc-gh-mnaides wrote keeps track the reason about why we are doing this, we won't document this option. But basically all the parameters/constants override should happen inside that class, feel free to suggest if you have a better idea

Reply:
I see. I wasnt aware of this. thank you! 

Lets just go with setParameterOverrides then without creating going through more changes.